### PR TITLE
Ensures that constants are ordered correctly in GUI.

### LINKED ISSUE #204: The constants are not correctly mapped in the rerun sim gui
The constants seem to be mixed up on load (try out the double pendulum example):
selection_087
g is set to 9.81 in the code and it gets mixed up with the other two in the JS side.

### PR DESCRIPTION
**Pull request by moorepants**
Fixes #204

I've tested this on the mass_spring_damper example and it works correctly now. There is also a ton of code clean up and new tests for Scene. I also removed the width and height attributes from Scene as they were not used anywhere.

- [x] There are no merge conflicts.
- [x] If there is a related issue, a reference to that issue is in the
  commit message.
- [x] Unit tests have been added for the bug. (Please reference the issue #
  in the unit test.)
- [x] The tests pass both locally (run `nosetests`) and on Travis CI.
- [x] The code follows PEP8 guidelines. (use a linter, e.g.
  [pylint](http://www.pylint.org), to check your code)
- [x] The bug fix is documented in the [Release
  Notes](https://github.com/pydy/pydy#release-notes).
- [x] The code is backwards compatible. (All public methods/classes must
  follow deprecation cycles.)
- [x] All reviewer comments have been addressed.

### CODE REVIEW

File: pydy/viz/scene.py [hunk 1]
```diff
@@ -367,15 +386,15 @@ def generate_visualization_json_system(self, system, **kwargs):
...
+            raise TypeError(msg)
         else:
-            #save system
             self.system = system
 
         self.generate_visualization_json(system.states,
-                                          system.constants_symbols,
-                                          system.integrate(),
-                                          system.constants.values(), **kwargs)
+                                         system.constants.keys(),
```
Comment (moorepants) [inline pydy/viz/scene.py]: This is the fix for the bug.

**Comment by moorepants** [comment]
@pydy/pydy-developers This needs a review. It is one small change to fix the bug and then I added basic tests for Scene.

**Comment by moorepants** [comment]
I'd like to get this merged ASAP. Can someone review?

File: pydy/viz/scene.py [hunk 2]
```diff
@@ -186,109 +153,104 @@ def generate_visualization_json(self, dynamic_variables,
...
+            self.generate_simulation_dict(dynamic_variables,
+                                          constant_variables,
+                                          dynamic_values,
+                                          constant_values)
 
+        self._scene_data_dict = \
+            self.generate_scene_dict(constant_map=constant_map)
 
+        self._scene_data_dict["simulationData"] = self.simulation_json_file
+        self._scene_data_dict["timeDelta"] = 1 / fps
```
Comment (chrisdembia) [inline pydy/viz/scene.py]: `int / int` won't be good; should this be `1.0 / fps` or `1 / float(fps)`?

File: pydy/viz/scene.py [hunk 3]
```diff
@@ -186,109 +153,104 @@ def generate_visualization_json(self, dynamic_variables,
...
-        simulation_data_outfile.close()
+        with open(self.scene_json_file, 'w') as scene_data_outfile:
+            scene_data_outfile.write(json.dumps(self._scene_data_dict,
+                                                indent=4,
+                                                separators=(',', ': ')))
+
+        with open(self.simulation_json_file, 'w') as simulation_data_outfile:
+            simulation_data_outfile.write(json.dumps(
+                self._simulation_data_dict, indent=4,
+                separators=(',', ': ')))
```
Comment (chrisdembia) [inline pydy/viz/scene.py]: +1

File: pydy/viz/scene.py [hunk 4]
```diff
@@ -186,109 +153,104 @@ def generate_visualization_json(self, dynamic_variables,
...
-        required by PyDy visualizer to draw the scene on the canvas.
+        """Generates a dictionary containing all of the information needed
+        to build the scene.
 
+        Parameters
+        ==========
+        constant_map : dictionary
+            A map of symbolic constants to numerical values. This is
+            typically used if there are symbolics in the dimensions of the
+            shapes.
```
Comment (chrisdembia) [inline pydy/viz/scene.py]: "dimensions of the shapes" can be confusing. At first I thought you meant like the dimensionality of a vector. Maybe "parameters of the shapes"? idk.

File: pydy/viz/scene.py [hunk 5]
```diff
@@ -186,109 +153,104 @@ def generate_visualization_json(self, dynamic_variables,
...
+
+        Returns
+        =======
+        scene_info : dictionary
```
Comment (chrisdembia) [inline pydy/viz/scene.py]: Is the formal type `dict` or `dictionary`?

File: pydy/viz/scene.py [hunk 6]
```diff
@@ -310,72 +271,98 @@ def generate_scene_dict(self, constant_map={}):
...
+        Notes
+        =====
 
-        Keyword arguments are same as generate_visualization_json.
+        The optional keyword arguments are same as the
+        ``generate_visualization_json`` method.
 
         """
         if not isinstance(system, System):
-            self.system = None
```
Comment (chrisdembia) [inline pydy/viz/scene.py]: Removing this doesn't change functionality? Are you aware of any resulting changes from deleting this line?

File: pydy/viz/tests/test_scene.py [hunk 7]
```diff
@@ -1,58 +1,243 @@
...
-    assert os.path.exists('static/index.html')
-    assert os.path.exists('static/test_scene_desc.json')
-    assert os.path.exists('static/test_simulation_data.json')
-
-    # test static dir deletion
-    scene.remove_static_html(force=True)
-    assert not os.path.exists('static')
+class TestScene(object):
+
+    def setup(self):
```
Comment (chrisdembia) [inline pydy/viz/tests/test_scene.py]: Perhaps a comment stating that this is a mass spring damper system?

**Comment by chrisdembia** [comment]
Please go ahead and merge after addressing my comments. The most important are the ones about `1 / fps` and about `self.system = None`.

File: pydy/viz/scene.py (hunk 2 above)
Comment (moorepants) [inline pydy/viz/scene.py]: `from future import division` is at the top of the file.

Comment (moorepants) [inline pydy/viz/scene.py]: Personally I prefer the explicitness of what you write though.

File: pydy/viz/scene.py (hunk 4 above)
Comment (moorepants) [inline pydy/viz/scene.py]: ok

File: pydy/viz/scene.py (hunk 5 above)
Comment (moorepants) [inline pydy/viz/scene.py]: I typically just write out the whole word. I can change them, but if you look at all my other code in PyDy I write it out.

File: pydy/viz/scene.py (hunk 2 above)
Comment (chrisdembia) [inline pydy/viz/scene.py]: Ahh I don't speak python3 yet.

File: pydy/viz/scene.py (hunk 6 above)
Comment (moorepants) [inline pydy/viz/scene.py]: Good question. The tests are so sparse, I have no idea. But I think I thought about it when I removed it. Will look more closely.

File: pydy/viz/tests/test_scene.py (hunk 7 above)
Comment (moorepants) [inline pydy/viz/tests/test_scene.py]: Sure.

**Comment by moorepants** [comment]
Thanks Chris. I'll fix it up.

File: pydy/viz/scene.py (hunk 6 above)
Comment (chrisdembia) [inline pydy/viz/scene.py]: Look at the blame.

File: pydy/viz/scene.py (hunk 5 above)
Comment (chrisdembia) [inline pydy/viz/scene.py]: leave as is

File: pydy/viz/scene.py (hunk 2 above)
Comment (moorepants) [inline pydy/viz/scene.py]: :smile: Me neither.

**Pull request merged**